File: ProjectEuler/Problem12b.c
#include <stdio.h>;
#include <math.h>;

int main() {
    int n = 1;
    int i = 1;
    while (1) {
        
        //don't perform test if it's not even (optimization)
        if (i % 2 == 0) {
        
            int divcount = 0;        
            int t;
            int t_limit = sqrt(i);
            for (t = 1; t <= t_limit; t++) {
                if (i % t == 0)
                    divcount += 2;
            }
            if (i == t_limit*t_limit)
                divcount--;
            
            if (divcount > 150) {
                printf("%d - %d\n", i, divcount);
            }
            
            if (divcount > 500) {
                printf("Found: %d - divcount %d\n", i, divcount);
                break;
            }
        }
        
        n++;
        i += n;
        
    }
}

File: ProjectEuler/Problem12b.py
import math

def triagen():
    n = i = 1
    while 1:
        yield i
        n += 1
        i += n
        
def divisor_count(n):
    divcount = 0
    tmax = math.sqrt(n)
    for t in xrange(1, tmax+1):
        if n % t == 0:
            divcount += 2
    if n == tmax:
        divcount -= 1
    return divcount
        
for i in triagen():
    divnum = divisor_count(i)
    if divnum > 150:
        print i, divnum
    if  divnum > 500:
        print i
        break